### ISSUE TITLE
Set up devcontainers

### ISSUE THREAD

**Issue by ehrhorn** [GlazeTech/TeraStore#29]
The VS Code workspace should include a devcontainers setup.

**Comment by ehrhorn** [comment]
I have started this on [29-set-up-devcontainers](https://github.com/GlazeTech/TeraStore/tree/29-set-up-devcontainers).

The reason being it seems smart to use a Dev Container for GitHub action workflows, see [here](https://github.com/devcontainers/ci).

To prevent too many "krumspring", I've toyed around with ditching Poetry, and using `pyproject.toml` in its standard sense. This is with `setuptools` as a backend. M'lord @Duckling92 may like it. Tho if I know him, he'll probably blast me for this experiment and want to revert to the olden times. Anyway, this makes it possible to install the package and dependencies with `pip install .`, and with the dev dependencies as `pip install .[test]`. Nice!

Issues:
* I am re-using the deployment `docker-compose.yml`, and installing the Python packages (incl. dev deps) in the `postCreateCommand` of `devcontainer.json`. That's fine, but it seems I need to point to which of the services in `docker-compose.yml` the Dev Container is based on. I'd like one Dev Container to go with both the frontend and backend environment, so have to look into that
* I don't understand the requirements stage of `docker/Dockerfile.backend`, so I don't know if it is relevant when Poetry is ditched
* My regex skills are weak, so I know that `include = '(?:tests|api)\/.*\.pyi?$'` for the Black config is wrong, I just don't know how to fix
* Mypy seems slow to start up at first, but then runs very fast
* VS Code has some timing issues... so when the Dev Container is rebuilt, it takes some time to install extensions (the required extensions are present in the `devcontainer.json` file), which leads to extensions that are already installed, but depend on others, to bitch. Bummer!

Anyway, this Dev Container, when ready, should be usable for CI and great for... well... development...
